xdsl/utils/diagnostic.py

```diff
@@ -30,6 +30,7 @@ def raise_exception(self, ir: IRNode, underlying_error: Exception) -> NoReturn:
                 p.print_string("\n")
             case Block():
                 p.print_block(toplevel)
+                p.print_string("\n")
```

Comment:
the last operation in a block is not terminated with a newline, so the diagnostic is not printed without this change.

Reply:
This might be left over from the `printer.print` changes